include/simdjson/generic/ondemand/object-inl.h

```diff
@@ -38,6 +38,11 @@ simdjson_really_inline simdjson_result<object> object::start(value_iterator &ite
   SIMDJSON_TRY( iter.start_object().get(has_value) );
   return object(iter);
 }
+simdjson_really_inline simdjson_result<object> object::start_root(value_iterator &iter) noexcept {
+  simdjson_unused bool has_value;
+  SIMDJSON_TRY( iter.start_root_object().get(has_value) );
```

Comment:
You do not have to do it, but a comment here explaining this construction might make the code easier to read.

Reply:
Oh! I agree. The "unused" being the weird bit?